onchain/wallet/wallet.go

```diff
@@ -754,11 +751,44 @@ func (wallet *Wallet) SendToAddress(address string, amount uint64, satPerVbyte f
 	})
 	defer free()
 
-	var result any
+	var result map[string]any
 	if err := withAuthHandler(C.GA_create_transaction(wallet.session, transactionDetails, handler), handler, &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (wallet *Wallet) GetSendFee(address string, amount uint64, satPerVbyte float64, sendAll bool) (send uint64, fee uint64, err error) {
+	result, err := wallet.createTransaction(address, amount, satPerVbyte, sendAll)
+	if err != nil {
+		return 0, 0, err
+	}
+	var createTx struct {
+		Fee        uint64
+		Addressees []struct {
+			Satoshi uint64
+		}
+		Error string
+	}
+	if err := mapstructure.Decode(result, &createTx); err != nil {
+		return 0, 0, err
+	}
+	if createTx.Error != "" {
+		return 0, 0, errors.New(createTx.Error)
+	}
+	return createTx.Addressees[0].Satoshi, createTx.Fee, nil
```

Comment:
Is it guaranteed that the first address is *always* the address we want to send to and not the change?

Reply:
This doesnt contain the actual outputs of the tx but only the addresses we told gdk to send to (gdk populates the amount in case of a sweep)